src/transformers/modeling_bert.py

```diff
@@ -957,7 +957,30 @@ def forward(
             ltr_lm_loss = loss_fct(prediction_scores.view(-1, self.config.vocab_size), lm_labels.view(-1))
             outputs = (ltr_lm_loss,) + outputs
 
-        return outputs  # (masked_lm_loss), (ltr_lm_loss), prediction_scores, (hidden_states), (attentions)
+        return outputs  # (ltr_lm_loss), (masked_lm_loss), prediction_scores, (hidden_states), (attentions)
+
+    def prepare_inputs_for_generation(self, input_ids, attention_mask=None, **model_kwargs):
+        # Add dummy token at the end (no attention on this one)
```

Comment:
I added now two cases here:
1) Bert is a decoder and therefore does casual masking. For this case, I assume it has been trained using the "normal" language modeling loss. Thus no need to add a dummy token.
2) Bert is **not** a decoder and therefore was trained using masked language modeling and needs to add a dummy token similar to how it's done for XLNet: https://github.com/huggingface/transformers/blob/778cf7cff6ffd32281b7583a0db0b566c0c76aee/src/transformers/modeling_xlnet.py#L941

What do you think @LysandreJik @thomwolf @sshleifer @yjernite ?